File: Csharp/ReadyToExam/ReadyToExam/NewYearGift/Model/Entity/Lollipop.cs
using System.Text;
using ReadyToExam.NewYearGift.Model.Abstracts;

namespace ReadyToExam.NewYearGift.Model.Entity
{
    public class Lollipop : Candy
    {
        private double _diameter;
        private string _taste;

        public Lollipop(double weight, string creator, double price, double diameter, string taste) : base(weight, creator, price)
        {
            _diameter = diameter;
            _taste = taste;
        }

        public double Diameter
        {
            get => _diameter;
            set => _diameter = value;
        }

        public string Taste
        {
            get => _taste;
            set => _taste = value;
        }
        
        public override string ToString()
        {
            return new StringBuilder().Append("Lolipop " + base.ToString() + "\nDiameter = " + _diameter + "\nTaste = " + _taste).ToString();
        }
    }
}

File: Java/Desktop/ReadyToExam/src/NewYearGift/Model/Entity/Lollipop.java
package NewYearGift.Model.Entity;

import NewYearGift.Model.Entity.Abstract.Candy;

public class Lollipop extends Candy {

    private double diameter;
    private String taste;

    public Lollipop(double weight, String creator, double price, double diameter, String taste) {
        super(weight, creator, price);
        this.diameter = diameter;
        this.taste = taste;
    }

    public double getDiameter() {
        return diameter;
    }

    public void setDiameter(double diameter) {
        this.diameter = diameter;
    }

    public String getTaste() {
        return taste;
    }

    public void setTaste(String taste) {
        this.taste = taste;
    }

    @Override
    public String toString(){
        return new StringBuilder().append("Lollipop ").append(super.toString()).append("Diameter: ").append(diameter).append("\nTaste: ").append(taste).toString();
    }

}
